src/data/structure/UserProfile.java

```diff
@@ -57,16 +57,21 @@ public String getId() {
      * @param preferenceId
      * @return added preference
      */
-    public Preference addPreference(PreferenceCategory category, String preferenceId) {
+    public Preference addPreference(PreferenceCategory category, String preferenceId, int popularity,
```

Comment:
For now, I made this a private method.  This way user contract for a user does not change.  Not sure that will be fully satisfactory, let me know.